src/compressed_tensors/utils/offload.py

```diff
@@ -99,8 +99,11 @@ def update_parameter_data(
         offloaded = True
 
     parameter = getattr(module, param_name, None)
-    dtype = parameter.dtype
-    parameter.data = new_param_data.to(device).to(dtype)
+    if parameter is not None:
+        dtype = parameter.dtype
+        parameter.data = new_param_data.to(device).to(dtype)
+    else:
+        setattr(module, param_name, new_param_data.to(device))
```

Comment:
I see, I agree that parameters ought to be initialized by `_initialize_scale_zero_point_observer`.
> The original intention was to ignore updates that didn't correspond to an existing parameter

If this is the case, then

Reply:
This case comes up when attempting to load a compressed model without `actorder`, as in this case the `weight_g_idx` parameter is not initialized.
```python3
if quantization_args.actorder:
    _, column_size = module.weight.shape
    init_g_idx = Parameter(
        torch.full((column_size,), -1, dtype=torch.int32, device=device),
        requires_grad=False,
    )
    module.register_parameter(f"{base_name}_g_idx", init_g_idx)
```

The original implementation throws a runtime error if `update_parameter_data` attempts to update the value of a missing parameter
```python3
parameter = getattr(module, param_name, None)
dtype = parameter.dtype
```

> Any needed quantization parameters should be initialized as Parameters at quantization initialization

This makes sense, I'll modify so that `weight_g_idx` is always initialized by `_initialize_scale_zero_point_observer` and have `update_parameter_data` throw a ValueError if misused on an uninitialized parameter